packages/core/src/components/fl-badge/fl-badge.tsx

```diff
@@ -0,0 +1,41 @@
+import { Component, Host, h, Prop, Element } from "@stencil/core";
+import { btypes, positions } from "./btypes";
+
+@Component({
+  tag: "fl-badge",
+  styleUrl: "fl-badge.scss",
+  shadow: true,
+})
+export class FlBadge {
+
+@Prop() type: btypes = btypes.PRIMARY;
+
+@Prop() pill = false;
+
+@Prop() position: positions = positions.NONE;
+
+/* stylelint-disable */
+@Element() private element: HTMLFlBadgeElement;
+/* stylelint-enable */
+
+componentWillLoad() {
+  this.element.parentElement.style.position = "relative";
+  // Just ensures some default type is set for badge
+  if (!Object.values(btypes).includes(this.type)) {
+    this.type = btypes.PRIMARY;
+  }
+}
+render() {
+  return (
+    <Host 
+      class={{
+        "badge": true,
+        [`badge--${this.type}`]: true,
+        [`badge--${this.position}`]: true,
+        "badge-pill": this.pill,
+      }}> {!this.element.innerHTML ? this.type : null}
```

Comment:
Do we want to render the type of the badge always? I think it'll be nice to customize this.

I think you can safely omit the `{!this.element.innerHTML ? this.type : null}`. The slot should be able to handle the rest

Reply:
Just made an updated branch with merges from main.